code/code/sys/socket.cc

```diff
@@ -561,42 +561,50 @@ void procMobHate::run(const TPulse &) const
 // decrease it by 1% (0.0046) per real day. until we arrive at 0.0460
 // but we'll do it per hour (.00019166666666666666)
 
-// procTweakLoadRate
-procTweakLoadRate::procTweakLoadRate(const int &p)
+// procTweakRate
+procTweakRate::procTweakRate(const int &p)
 {
   trigger_pulse=p;
-  name="procTweakLoadRate";
+  name="procTweakRate";
 }
 
-// The Target loadrate and rate of change should be set from within the mud,
+// The Target tweak and rate of change should be set from within the mud,
 // not hard-coded
-void procTweakLoadRate::run(const TPulse &) const
+void procTweakRate::run(const TPulse &) const
 {
+  TDatabase db(DB_SNEEZY);
 
-  if ( stats.global_lp_target_changerate <= 0 ) return;     // If this is zero, the procedure won't change anything anyway.
 
+  for(tweakTypeT i=TWEAK_LOADRATE;i<MAX_TWEAK_TYPES;i++){
+    
+    tweakEntry* t = tweakInfo[i];
+    if (t->rate == 0.0) {
+      continue;
+    }
 
-  // target_delta = (stats.global_lp_target - original_load)/(24hr/day*3600s/hr*delta_time_in_days)
+    if ( t->tvalue <= 0 ) return;     // If this is zero, the procedure won't change anything anyway.
```

Comment:
... and this style could be unified a bit. Also, the extra spaces inside `()` don't follow the majority of style